assets/src/edit-story/components/library/panes/shared/libraryMoveable.js

```diff
@@ -0,0 +1,260 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useCallback, useRef, useState } from 'react';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { editorToDataX, editorToDataY } from '../../../../units';
+import Moveable from '../../../moveable';
+import useSnapping from '../../../canvas/utils/useSnapping';
+import { useDropTargets } from '../../../dropTargets';
+import { useLayout } from '../../../../app/layout';
+import useInsertElement from '../../../canvas/useInsertElement';
+import { useCanvas } from '../../../canvas';
+import isMouseUpAClick from '../../../../utils/isMouseUpAClick';
+import InOverlay from '../../../overlay';
+import isTargetOutOfContainer from '../../../../utils/isTargetOutOfContainer';
+import { useGlobalKeyDownEffect } from '../../../keyboard';
+import { useStory } from '../../../../app/story';
+import objectWithout from '../../../../utils/objectWithout';
+
+const TargetBox = styled.div`
+  position: absolute;
+  width: ${({ width }) => `${width}px`};
+  height: ${({ height }) => `${height}px`};
+`;
+
+function LibraryMoveable({
+  type,
+  elementProps = {},
+  handleDrag,
+  handleDragEnd,
+  onClick,
+  cloneElement,
+  cloneProps,
+  active = false,
+}) {
+  const CloneElement = cloneElement;
+
+  const [isDragging, setIsDragging] = useState(false);
+  const [didManuallyReset, setDidManuallyReset] = useState(false);
+  const cloneRef = useRef(null);
+  const targetBoxRef = useRef(null);
+  const overlayRef = useRef(null);
+
+  const { pageSize } = useLayout(({ state }) => ({
+    pageSize: state.canvasPageSize,
+  }));
+
+  const insertElement = useInsertElement();
+  const { pageContainer, nodesById } = useCanvas((state) => ({
+    pageContainer: state.state.pageContainer,
+    nodesById: state.state.nodesById,
+  }));
+
+  const {
+    state: { activeDropTargetId },
+    actions: { setDraggingResource },
+  } = useDropTargets();
+
+  const { backgroundElement } = useStory(({ state: { currentPage } }) => ({
+    backgroundElement: currentPage?.elements?.[0] ?? {},
+  }));
+
+  const frame = {
+    translate: [0, 0],
+  };
+
+  const eventTracker = useRef({});
+  const startEventTracking = (evt) => {
+    const { timeStamp, clientX, clientY } = evt;
+    eventTracker.current = {
+      timeStamp,
+      clientX,
+      clientY,
+    };
+  };
+
+  const resetMoveable = useCallback(() => {
+    targetBoxRef.current.style.transform = null;
+    cloneRef.current.style.transform = null;
+    // Hide the clone, too.
+    cloneRef.current.style.opacity = 0;
+    setIsDragging(false);
+    setDraggingResource(null);
+  }, [setDraggingResource]);
+
+  useGlobalKeyDownEffect(
```

Comment:
This is actually interfering with other esc handlers and can't be used here...

Tried using `useKeyDownEffect` -- checking the `document.activeElement` while dragging then it's `document.body` -- however, this would also interfere with the global `esc` handlers so not sure what would be the approach here to implement the keyhandler. 🤔

Reply:
Found a potential fix for this.